packages/injectable/extension-for-mobx/src/computedInjectMany.test.js

```diff
@@ -398,4 +398,111 @@ describe('registerMobx', () => {
       });
     });
   });
+
+  describe("given the MobX extension is registered, and there is complex injection token and implementations of it, when injected as reactive", () => {
+    let di;
+    let actual;
+    let someFirstInjectionToken;
+    let someOtherInjectable;
+    let reactionCountForFirstToken;
+    let someInjectable;
+    let contextsOfSomeInjectable;
+
+    beforeEach(() => {
+      configure({
+        enforceActions: "never",
+        computedRequiresReaction: false,
+        observableRequiresReaction: false,
+      })
+      contextsOfSomeInjectable = [];
+      reactionCountForFirstToken = 0;
+
+      someFirstInjectionToken = getInjectionToken({
+        id: 'some-injection-token',
+      });
+
+      someInjectable = getInjectable({
+        id: 'some-injectable',
+        instantiate: () => observable({
+          someField: 'some-instance',
+        }),
+        injectionToken: someFirstInjectionToken,
+      });
+
+      const contextSpyDecorator = getInjectable({
+        id: 'context-spy-decorator',
+
+        instantiate: () => ({
+          target: someInjectable,
+
+          decorate:
+            toBeDecorated =>
+              (alias, instantiationParameter, context = []) => {
+                contextsOfSomeInjectable.push([
+                  ...context.map(x => x.injectable.id),
+                  alias.id,
+                ]);
+
+                return toBeDecorated(alias, instantiationParameter, context);
+              },
+        }),
+
+        decorable: false,
+
+        injectionToken: injectionDecoratorToken,
+      });
+
+      someOtherInjectable = getInjectable({
+        id: 'some-other-injectable',
+        instantiate: () => observable({
+          someField: 'some-other-instance',
+        }),
+        injectionToken: someFirstInjectionToken,
+      });
+
+      di = createContainer('some-container');
+
+      di.register(contextSpyDecorator);
+
+      registerMobX(di);
+    });
+
+    describe('given in reactive context and observed as computedInjectMany, when multiple injectables that implement the injection token are registered', () => {
+      beforeEach(() => {
+        const computedInjectMany = di.inject(computedInjectManyInjectable);
+
+        actual = computedInjectMany(someFirstInjectionToken);
+
+        runInAction(() => {
+          di.register(someInjectable, someOtherInjectable);
+        });
+      });
+
+      it('injects reactive instances', () => {
+        expect(actual.get()).toStrictEqual([
```

Comment:
Accessing value without reactive context seems weird and can cause trouble here. I would like to see the test written by using `reaction` 

By using that you'll see that you will need to observe the individual values (`someField` in this case) to make it work. This seems to be same behaviour than using `observable.array` instead of `computedInjectMany`

**Minimal example using `observable.array`**

```typescript
const someObservable = observable({ someProperty: 'some-value' });

const someObservableArray = observable.array([someObservable]);

let actual;

reaction(
  // Notice: someProperty is dereferenced
  () => someObservableArray.map(x => x.someProperty),
  changed => {
    actual = changed;
  },
  { fireImmediately: true },
);

expect(actual).toEqual(['some-value']);

runInAction(() => {
  someObservable.someProperty = 'some-changed-value';
});

expect(actual).toEqual(['some-changed-value']);
```

Notice how changing the `reaction` a little bit would break the test:

```typescript
reaction(
  // Notice: no dereferencing is done
  () => someObservableArray,
  changed => {
    actual = changed.map(x => x.someProperty);
  },
  { fireImmediately: true },
);
```